service/header/sync.go

```diff
@@ -10,62 +11,119 @@ import (
 	tmbytes "github.com/tendermint/tendermint/libs/bytes"
 )
 
+// TODO:
+//  1. Sync protocol for peers to exchange their local heads on connect
+//  2. If we are far from peers but within an unbonding period - trigger sync automatically
+//  3. If we are beyond the unbonding period - request Local Head + 1 header from trusted and hardcoded peer
+//  automatically and continue sync until know head.
+//  4. Limit amount of requests on server side
+//  5. Sync status and till sync is done.
+//  6. Cache even unverifiable headers from the future, but don't resend them with ignore.
+//  7. Retry requesting headers
+
 // Syncer implements simplest possible synchronization for headers.
 type Syncer struct {
+	sub      *P2PSubscriber
 	exchange Exchange
 	store    Store
 	trusted  tmbytes.HexBytes
 
 	// inProgress is set to 1 once syncing commences and
 	// is set to 0 once syncing is either finished or
 	// not currently in progress
-	inProgress uint64
+	inProgress  uint64
```

Comment:
Is there a reason to not use a `bool` here?

Reply:
could use an atomic.Value i guess?

Reply:
There is no std atomic bool. Using int as bool is idiomatic.

Reply:
Some people using https://github.com/uber-go/atomic which provides an atomic bool, which is int underneath